nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -154,6 +157,13 @@ void RegulatedPurePursuitController::configure(
     use_cost_regulated_linear_velocity_scaling_ = false;
   }
 
+  // Possible to drive in reverse direction if and only if "use_rotate_to_heading" parameter is set to false
+
+  if(use_rotate_to_heading_)
+  {
```

Comment:
This will fail linting, please follow the ROS2 style guide and run tests over code before submitting PRs. This can be caught locally by doing so or running ament_cpplint and ament_uncrustify

Reply:
Why is this even related to rotating to heading? It shouldn't be

Reply:
Yeah I think that's a good call. It might be appropriate at times to use both. Though it would be worth testing on your side if you have both set what actually happens at cusping events

Reply:
I gave it a go and when I set both the params to true, all the time it rotates to the heading and goes forward. Here reversing is not considered at all.

Reply:
Got it OK. Then I think you are correct to have these be mutually exclusive. Please update the readme and navigation.ros.org configuration page (https://github.com/ros-planning/navigation.ros.org/blob/master/configuration/packages/configuring-regulated-pp.rst) to mention that in those parameterizations. We should also check on initialization if both are set to true to throw an error saying that is an invalid configuration, only 1 can be on at a time. Then default to rotating in place (since I think of the 2, that's the "safer" default). Do you agree?

Reply:
> We should also check on initialization if both are set to true to throw an error saying that is an invalid configuration, only 1 can be on at a time.

That would indeed be a fail proof 

> Then default to rotating in place (since I think of the 2, that's the "safer" default). Do you agree?

Sure, I agree to this point. I will update the documentation in parallel to the changes we discussed here ✌🏽

Reply:
Thanks for this great work! It will be great to have!

Reply:
No problem, thank you for the guidance 😃

Reply:
> We should also check on initialization if both are set to true to throw an error saying that is an invalid configuration, only 1 can be on at a time. Then default to rotating in place (since I think of the 2, that's the "safer" default).

Here maybe, we could just throw a warn message stating that, both the parameters are set to true, so we are setting the default to rotating in place ?

Reply:
yes